x-pack/solutions/security/plugins/security_solution/server/endpoint/services/agent/clients/microsoft_defender_endpoint/microsoft_defender_endpoint_agent_status_client.ts

```diff
@@ -66,22 +70,78 @@ export class MicrosoftDefenderEndpointAgentStatusClient extends AgentStatusClien
     }
   }
 
+  protected async calculateHostIsolatedState(agentIds: string[]): Promise<Record<string, boolean>> {
+    const response: Record<string, boolean> = {};
+    const errors: string[] = [];
+
+    await pMap(
+      agentIds,
+      async (agentId) => {
+        response[agentId] = false;
+
+        try {
+          // Microsoft's does not seem to have a public API that enables us to get the Isolation state for a machine. To
+          // get around this, we query the list of machine actions for each host and look at the last successful
+          // Isolate or Unisolate action to determine if host is isolated or not.
+          const { data: hostLastSuccessfulMachineAction } = await this.connectorActions.execute<
+            MicrosoftDefenderEndpointGetActionsResponse,
+            MicrosoftDefenderEndpointGetActionsParams
+          >({
+            params: {
+              subAction: MICROSOFT_DEFENDER_ENDPOINT_SUB_ACTION.GET_ACTIONS,
+              subActionParams: {
+                status: 'Succeeded',
+                type: ['Isolate', 'Unisolate'],
+                pageSize: 1,
+                sortField: 'lastUpdateDateTimeUtc',
```

Comment:
nice workaround!

is it possible that an action entry is updated long after it's succeeded?
e.g.
- `action1` isolate performed at `t1` and succeeded at `t2`
- `action2` unisolate performed at `t3` and succeeded at `t4`
- for some reason, `action1` is updated at `t5`
- hence, sorting by `lastUpdateDateTimeUtc` will return with `action1` isolate and we think that the host is isolated

do you think this is possible?

Reply:
I don't know for sure if thats possible since I we don't have insight into MS's business logic behind these "Machine actions". Personally, I don't think that sounds like a valid use case since the action was already "completed".